presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlock.java

```diff
@@ -303,9 +308,48 @@ public Block getLoadedBlock()
                 offsets,
                 keyBlock,
                 loadedValueBlock,
-                hashTables,
+                Optional.ofNullable(hashTables),
                 keyType,
                 keyBlockNativeEquals,
-                keyNativeHashCode);
+                keyNativeHashCode,
+                keyBlockHashCode);
+    }
+
+    @Override
+    protected void ensureHashTableLoaded()
+    {
+        if (getHashTables() == null) {
```

Comment:
Why not access `this.hashTables` directly? you do this when you complete computation anyway